File: services/device/src/main/java/com/example/iter/device/domain/repository/EquipmentImageRepository.java
package com.example.iter.device.domain.repository;

import com.example.iter.device.domain.entity.EquipmentImage;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Collection;
import java.util.List;

public interface EquipmentImageRepository extends JpaRepository<EquipmentImage, Long> {
    List<EquipmentImage> findByEquipmentIdOrderBySortOrderAsc(Long equipmentId);

    List<EquipmentImage> findByEquipmentIdOrderBySortOrderAscIdAsc(Long equipmentId);

    // 여러 장비의 썸네일을 한 번에 조회합니다.
    List<EquipmentImage>
    findByEquipment_IdInAndThumbnailTrueOrderBySortOrderAscIdAsc(Collection<Long> equipmentIds);
}

File: services/device/src/main/kotlin/com/example/iter/device/domain/repository/EquipmentImageRepository.kt
package com.example.iter.device.domain.repository

import com.example.iter.device.domain.entity.EquipmentImage
import org.springframework.data.jpa.repository.JpaRepository

interface EquipmentImageRepository : JpaRepository<EquipmentImage, Long> {

    fun findByEquipmentIdOrderBySortOrderAsc(equipmentId: Long): List<EquipmentImage>

    fun findByEquipmentIdOrderBySortOrderAscIdAsc(equipmentId: Long): List<EquipmentImage>

    // 여러 장비의 썸네일을 한 번에 조회합니다.
    fun findByEquipment_IdInAndThumbnailTrueOrderBySortOrderAscIdAsc(
        equipmentIds: Collection<Long>,
    ): List<EquipmentImage>
}
